readthedocs/projects/forms.py

```diff
@@ -621,11 +621,19 @@ def clean_project(self):
         return self.project
 
     def clean_domain(self):
-        parsed = urlparse(self.cleaned_data['domain'])
-        if parsed.scheme or parsed.netloc:
-            domain_string = parsed.netloc
-        else:
-            domain_string = parsed.path
+        domain = self.cleaned_data['domain']
+        parsed = urlparse(domain)
+
+        # Force the scheme to have a valid netloc.
+        if not parsed.scheme:
+            parsed = urlparse(f'https://{domain}')
+
+        if not parsed.netloc:
```

Comment:
This is much better than nothing, but it isn't a great check:

```
In [13]: parse.urlparse('https://*****.laksdhg.foo')
Out[13]: ParseResult(scheme='https', netloc='*****.laksdhg.foo', path='', params='', query='', fragment='')
```

Reply:
Looks like something the domain validator should be catching https://github.com/readthedocs/readthedocs.org/blob/b74fe8d5aab8421fb194d7b8cbbb42a4e54f2569/readthedocs/projects/validators.py#L15-L24

But isn't... Somehow that regex is allowing that value, I took at look at the new version of that validator https://github.com/django/django/pull/7300/files but the result is the same. Also, I think we shouldn't allow ip values here